src/Slices/Dashboard/UI/GraphCard.tsx

```diff
@@ -27,40 +27,38 @@ export const GraphCard: React.FC<GraphCardProps> = ({
 }) => {
   const [chartState, setChartState] = useState<Metric[]>([]);
   const [legendDataState, setLegendDataState] = useState<LegendData[]>([]);
-  const [minMax, setMinMax] = useState<number[]>([]);
+  const [max, setMax] = useState<number>(0);
 
   useEffect(() => {
     if (isStacked) {
-      const tempCharState: Metric[] = [
-        {
-          name: "down",
-          data: [],
-        },
-        {
-          name: "paused",
-          data: [],
-        },
-        {
-          name: "up",
-          data: [],
-        },
-      ];
+      const tempCharState: Metric[] = [];
+      let max = 0;
       const { data } = metrics as StackedMetric;
-      data.map(({ down, paused, up }) => {
-        tempCharState[0].data.push(down);
-        tempCharState[1].data.push(paused);
-        tempCharState[2].data.push(up);
-      });
-      const dataValuesOnly = tempCharState.map((object) => object.data);
-      const tempMinMax = dataValuesOnly[0]
-        .map(
-          (value, index) =>
-            value + dataValuesOnly[1][index] + dataValuesOnly[2][index]
-        )
-        .sort((a, b) => a - b);
+      const base = data.find((object) => object !== null);
+      if (base !== undefined && base !== null) {
+        const keys = Object.keys(base);
+        keys.map((key) => {
+          tempCharState.push({
+            name: key,
+            data: [],
+          });
+        });
+        data.map((object) => {
+          let tempMax = 0;
+          keys.forEach((key, index) => {
+            tempMax += object === null ? 0 : object[key];
+            tempCharState[index].data.push(
+              object === null ? null : object[key]
```

Comment:
`object[key] || null` ?

Reply:
Doesn't work, I got an error, both by typescript and in the app

Reply:
ok!